packages/taco/src/conditions/base/shared.ts

```diff
@@ -1,11 +1,47 @@
 import { z } from 'zod';
 
-import { ETH_ADDRESS_REGEXP, USER_ADDRESS_PARAM } from '../const';
+import {
+  CONTEXT_PARAM_REGEXP,
+  ETH_ADDRESS_REGEXP,
+  USER_ADDRESS_PARAM,
+} from '../const';
+import { CONTEXT_PARAM_PREFIX } from '../context/context';
+
+export const contextParamSchema = z.string().regex(CONTEXT_PARAM_REGEXP);
+// We want to discriminate between ContextParams and plain strings
+// If a string starts with `:`, it's a ContextParam
+export const plainStringSchema = z.string().refine(
+  (str) => {
+    if (str.startsWith(CONTEXT_PARAM_PREFIX)) {
+      return str.match(CONTEXT_PARAM_REGEXP);
+    }
+    return true;
+  },
+  {
+    message: 'Context parameters must start with ":"',
+  },
```

Comment:
Does this mean that a string that starts with `:` and matches the context regex is considered a valid plain string? That doesn't seem right, but it's possible I don't understand the goal of this schema.

Reply:
You understood correctly - This schema will accept a plain string or a condition context param, so it should have been called `plainStringOrContextParamSchema`.

This was unintended. At the time of writing, I thought I needed a negative statement to validate two different types of string, so I added that edge case. But it turns out it's much simpler, I'm going to publish an update shortly.